### PR TITLE
feature: loglevel value parser

### LINKED ISSUE #8465: remove clap feature from LogLevel
### Describe the feature

enabling features for crates deep in the stack is very inconvenient

https://github.com/paradigmxyz/reth/blob/fe823cba7eb609e62505812a0fc057f2bf41b7ed/crates/storage/errors/src/db.rs#L103-L103

This feature is only used here

https://github.com/paradigmxyz/reth/blob/fe823cba7eb609e62505812a0fc057f2bf41b7ed/crates/node-core/src/args/database.rs#L11-L13

## TODO
* remove clap support and instead implement a value parser that manually does this

this is the generated code

```
impl clap::ValueEnum for LogLevel {
        fn value_variants<'a>() -> &'a [Self] {
            &[
                Self::Fatal,
                Self::Error,
                Self::Warn,
                Self::Notice,
                Self::Verbose,
                Self::Debug,
                Self::Trace,
                Self::Extra,
            ]
        }
        fn to_possible_value<'a>(
            &self,
        ) -> ::std::option::Option<clap::builder::PossibleValue> {
            match self {
                Self::Fatal => {
                    Some({
                        clap::builder::PossibleValue::new("fatal")
                            .help(
                                "Enables logging for critical conditions, i.e. assertion failures",
                            )
                    })
                }
                Self::Error => {
                    Some({
                        clap::builder::PossibleValue::new("error")
                            .help("Enables logging for error conditions")
                    })
                }
                Self::Warn => {
                    Some({
                        clap::builder::PossibleValue::new("warn")
                            .help("Enables logging for warning conditions")
                    })
                }
                Self::Notice => {
                    Some({
                        clap::builder::PossibleValue::new("notice")
                            .help("Enables logging for normal but significant condition")
                    })
                }
                Self::Verbose => {
                    Some({
                        clap::builder::PossibleValue::new("verbose")
                            .help("Enables logging for verbose informational")
                    })
                }
                Self::Debug => {
                    Some({
                        clap::builder::PossibleValue::new("debug")
                            .help("Enables logging for debug-level messages")
                    })
                }
                Self::Trace => {
                    Some({
                        clap::builder::PossibleValue::new("trace")
                            .help("Enables logging for trace debug-level messages")
                    })
                }
                Self::Extra => {
                    Some({
                        clap::builder::PossibleValue::new("extra")
                            .help("Enables logging for extra debug-level messages")
                    })
                }
                _ => None,
            }
        }
    }
```

this can also be solved by copying the `value_variants` and add a `fn help_message` which the value parser then uses

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by JackG-eth**
custom clap value parser for LogLevel enum

Motivation
remove clap feature from LogLevel: closes #8465

### CODE REVIEW

File: crates/storage/errors/src/db.rs [hunk 1]
```diff
@@ -122,3 +126,88 @@ pub enum LogLevel {
...
+            _ => Err(format!("Invalid log level: {}", s)),
+        }
+    }
+}
+
+/// clap value parser for [`LogLevel`].
+#[derive(Clone, Debug, Default)]
+pub struct LogLevelValueParser;
+
+impl TypedValueParser for LogLevelValueParser {
```
Comment (mattsse) [inline crates/storage/errors/src/db.rs:182]: this should be moved to node-core

**Review by mattsse — CHANGES_REQUESTED** [review]
nice, valueparser looks great

now we need to move this to node-core so we can actually get rid of the clap dep

File: crates/storage/errors/src/db.rs (hunk 1 above)
Comment (JackG-eth) [reply crates/storage/errors/src/db.rs]: of course, mb

File: crates/node-core/src/args/database.rs [hunk 2]
```diff
@@ -26,6 +30,42 @@ impl DatabaseArgs {
     }
 }
 
+/// clap value parser for [`LogLevel`].
+#[derive(Clone, Debug, Default)]
+struct LogLevelValueParser;
```
Comment (mattsse) [inline crates/node-core/src/args/database.rs:35]:
```suggestion
#[derive(Clone, Debug, Default)]
#[non_exhaustive]
pub struct LogLevelValueParser;
```

File: crates/node-core/src/args/database.rs [hunk 3]
```diff
@@ -26,6 +30,42 @@ impl DatabaseArgs {
...
+                "Invalid value '{val}' for {arg}: {err}.\n    Possible values:\n{possible_values}"
+            );
+            clap::Error::raw(clap::error::ErrorKind::InvalidValue, msg)
+        })
+    }
+
+    fn possible_values(&self) -> Option<Box<dyn Iterator<Item = PossibleValue> + '_>> {
+        let values =
+            LogLevel::value_variants().iter().map(|v| PossibleValue::new(v.help_message()));
+        Some(Box::new(values))
```
Comment (mattsse) [inline crates/node-core/src/args/database.rs:66]: this needs the same code as in the issue ` fn to_possible_value` with possiblevalue name + help

**Review by mattsse — CHANGES_REQUESTED** [review]
almost there

**Review by mattsse — APPROVED** [review]
ty

**Pull request merged**